src/core/font_renderer.js

```diff
@@ -674,7 +688,9 @@ var FontRendererFactory = (function FontRendererFactoryClosure() {
     this.cmap = cmap;
     this.glyphNameMap = glyphNameMap || getGlyphsUnicode();
 
-    this.compiledGlyphs = [];
+    this.compiledGlyphs = Object.create(null);
+    this.compiledCharCodeToGlyphId = Object.create(null);
```

Comment:
Comment https://github.com/mozilla/pdf.js/pull/7106/files#r57228709 should apply here too.